juju/juju
# [JUJU-314] Fix maas bond interface bug

Maas provider assumes bond interfaces start with 'bond'. This is not always the case though

https://bugs.launchpad.net/juju/+bug/1692611

## QA steps

- Set up a MAAS node with bonded NICs; name the bond anything not starting with "bond".
- `juju model-config -m controller logging-config="<root>=INFO;juju.apiserver.instancepoller=TRACE"`
- Add the MAAS node as a Juju machine (via constraints or placement).
- `juju debug-log -m controller`.
- Observe that the bonded NICs correctly have their parent set to the bond.
